src/test/java/org/wildfly/security/authz/RoleMappingTest.java

```diff
@@ -0,0 +1,253 @@
+                .setRemoveNonMapped(true)
+                .setRoleMap(mappingMap).build();
+
+        Roles mappedRoles = mapper.mapRoles(roles);
+
+        assertTrue(mappedRoles.contains("bar"));
+        assertTrue(mappedRoles.contains("role"));
+        assertFalse(mappedRoles.contains("foo"));
+        assertFalse(mappedRoles.contains("joe"));
+
+        Iterator<String> iterator = mappedRoles.iterator();
+        int count = 0;
+        while (iterator.hasNext()) {
+            iterator.next();
+            count++;
+        }
+        assertTrue(count == 2);
+    }
+
+    @Test
+    public void testMappedRoles2() {
+        Roles roles = createRoles("foo", "joe");
+
+        Map<String, Set<String>> mappingMap = new HashMap<>();
+        mappingMap.put("foo", createSet("bar", "role"));
+
+        RoleMapper mapper = new MappedRoleMapper.Builder()
+                .setRemoveMapped(false)
+                .setRemoveNonMapped(true)
+                .setRoleMap(mappingMap).build();
+
+        Roles mappedRoles = mapper.mapRoles(roles);
+
+        assertTrue(mappedRoles.contains("bar"));
+        assertTrue(mappedRoles.contains("role"));
+        assertTrue(mappedRoles.contains("foo"));
+        assertFalse(mappedRoles.contains("joe"));
+
+        Iterator<String> iterator = mappedRoles.iterator();
+        int count = 0;
+        while (iterator.hasNext()) {
+            iterator.next();
+            count++;
+        }
+        assertTrue(count == 3);
+    }
+
+
+    @Test
+    public void testMappedRoles3() {
+        Roles roles = createRoles("foo", "joe");
+
+        Map<String, Set<String>> mappingMap = new HashMap<>();
+        mappingMap.put("foo", createSet("bar", "role"));
+
+        RoleMapper mapper = new MappedRoleMapper.Builder()
+                .setRemoveMapped(true)
+                .setRemoveNonMapped(false)
+                .setRoleMap(mappingMap).build();
+
+        Roles mappedRoles = mapper.mapRoles(roles);
+
+        assertTrue(mappedRoles.contains("bar"));
+        assertTrue(mappedRoles.contains("role"));
+        assertFalse(mappedRoles.contains("foo"));
+        assertTrue(mappedRoles.contains("joe"));
+
+        Iterator<String> iterator = mappedRoles.iterator();
+        int count = 0;
+        while (iterator.hasNext()) {
+            iterator.next();
+            count++;
+        }
+        assertTrue(count == 3);
+    }
+
+    @Test
+    public void testSuffixRoles() {
+        String[] stringRoles = {"foo", "bar"};
+        String suffix = "suffix";
+
+        Roles roles = createRoles(stringRoles);
+
+        Roles suffixRoles = roles.addSuffix(suffix);
+
+        for (String s : stringRoles) {
+            assertTrue(suffixRoles.contains(s + suffix));
+        }
+
+        int count = 0;
+        for (Iterator<String> iterator = suffixRoles.iterator(); iterator.hasNext(); ) {
+            iterator.next();
+            count++;
+        }
+        assertTrue(count == 2);
```

Comment:
usually we use assertEquals ;)

Reply:
alright, will fix that

Reply:
@hkalina wait, isn't Assert.assertArrayEquals index dependent?